groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -218,40 +221,389 @@ extern const Optional_OptNoSuchType optNoSuchType;
+
+#define BSLSTL_OPTIONAL_ENABLE_ASSIGN_FROM(TYPE, ANY_TYPE)                    \
+    typename bsl::enable_if<                                                  \
+        !bsl::is_same<ANY_TYPE, optional<TYPE> >::value &&                    \
+            !(bsl::is_same<ANY_TYPE,                                          \
+                           typename bsl::decay<TYPE>::type>::value &&         \
+              std::is_scalar<TYPE>::value) &&                                 \
+            BloombergLP::bslstl::                                             \
+                Optional_IsConstructible<TYPE, ANY_TYPE, true>::value &&      \
+            BloombergLP::bslstl::                                             \
+                Optional_IsAssignable<TYPE&, ANY_TYPE, true>::value,          \
+        optional<TYPE> >::type
+
+#define BSLSTL_OPTIONAL_ENABLE_IF_NOT_ALLOCATOR_TAG(ARG)                      \
+    typename bsl::enable_if<                                                  \
+        !bsl::is_same<                                                        \
+            typename BloombergLP::bslstl::Optional_RemoveCVRef<ARG>::type,    \
+            bsl::allocator_arg_t>::value,                                     \
+        optional<TYPE> >::type
+
                            // ======================
                            // class Optional_DataImp
                            // ======================
 
 template <class TYPE>
 struct Optional_DataImp {
-    // This component-private 'struct' manages a 'value_type' object in
-    // 'optional'.  This class provides an abstraction for 'const' value type.
-    // An 'optional' may contain a 'const' type object.  An assignment to such
-    // an 'optional' should not succeed.  However, unless the 'optional' itself
-    // is 'const', it should be possible to change the value of the 'optional'
-    // using 'emplace'.  In order to allow for that, this class manages a
-    // non-const object of 'value_type', but all the accessors return a 'const'
-    // adjusted reference to the managed object.
+    // This component-private 'struct' manages a 'value_type' object in an
+    // 'optional' object.  This class provides an abstraction for 'const' value
+    // type.  An 'optional' object may contain an object of 'const' type.  An
+    // assignment to such an 'optional' object should not succeed.  However,
+    // unless the 'optional' object itself is 'const', it should be possible to
+    // change the value of the 'optional' object using 'emplace'.  In order to
+    // allow for that, this class manages a non-const object of 'value_type',
+    // but all the accessors return a 'const' adjusted reference to the managed
+    // object.
 
   private:
     // PRIVATE TYPES
     typedef typename bsl::remove_const<TYPE>::type StoredType;
 
     // DATA
-    bsls::ObjectBuffer<StoredType> d_buffer;
+    BloombergLP::bsls::ObjectBuffer<StoredType> d_buffer;
```

Comment:
```suggestion
    bsls::ObjectBuffer<StoredType> d_buffer;
```
Isn't this struct in `BloombergLP` namespace?  Or am I missing something?

Reply:
it is. removed Bloomberg namespace here and in a similar instance below